### ISSUE TITLE
Long Question

### ISSUE THREAD

**Issue by TopZoozle** [PrismarineJS/mineflayer#1782]
Was having a lot of trouble with this...

I've read a lot of different methods of doing it and I just want to know whether someone here can give me a concrete answer.

I want to run multiple bots (yes I know this is possible and have already done it)
I want to run these bots from a file (have already done this)
I want to control the bots via discord (can do in a single instance but not when running multiple bots)
I want to log the bots in via proxy (have done this in a single instance) but I want to do it through multiple. Any help would be greatly appreciated.

**Comment by Moondarker** [comment]
This question doesn't really contain any questions

**Comment by TopZoozle** [comment]
Ok let me rephrase, How would I log into multiple bots from a file while running the bots through proxy once again from a file.

**Comment by u9g** [comment]
> Ok let me rephrase, How would I log into multiple bots from a file while running the bots through proxy once again from a file?

cant you just add the proxy to the creatbot?

**Comment by TopZoozle** [comment]
> > Ok let me rephrase, How would I log into multiple bots from a file while running the bots through proxy once again from a file?
> 
> cant you just add the proxy to the creatbot?

In a single instance yes, running it through a file isn't that simple, unfortunately.

**Comment by u9g** [comment]
> > > Ok let me rephrase, How would I log into multiple bots from a file while running the bots through proxy once again from a file?
> > 
> > 
> > cant you just add the proxy to the creatbot?
> 
> In a single instance yes, running it through a file isn't that simple, unfortunately.

what is your code rn?

**Comment by TopZoozle** [comment]
haven't been able to implement adding the proxies into the same map as the login.
`const fs = require('fs');
const util = require('util');
const mineflayer = require('mineflayer');
const readFile = (fileName) => util.promisify(fs.readFile)(fileName, 'utf8');
const { pathfinder, Movements, goals: { GoalNear } } = require('mineflayer-pathfinder');
const Socks = require('socks5-client');
const RANGE_GOAL = 1;

const config = {
    host: 'localhost',
    port: '25565',
    file: './accounts.txt',
    proxyfile: './socks4.txt',
    interval: 500
}
var i;

function makeBot([_u, _p], ix) {
    return new Promise((resolve, reject) => {
        setTimeout(() => {

            const bot = mineflayer.createBot({
                host: config.host,
                port: config.port,
                username: _u,
                password: _p,

            });
            console.log('Attempting To Connect');
            bot.once('spawn', () => {

                bot.loadPlugin(pathfinder);
                const mcData = require('minecraft-data')(bot.version);
                const defaultMove = new Movements(bot, mcData);
                defaultMove.allowFreeMotion = true;
                bot.pathfinder.searchRadius = 100;
                defaultMove.allowParkour = true;
                bot.on('chat', (username, message) => {
                    if (username === bot.username) return;
                    if (message === 'follow') {
                        const target = bot.players[username].entity;

                    }
                    if (message === 'register') {

                    }
                    if (message === 'login') {

                    }
                    const { x: playerX, y: playerY, z: playerZ } = target.position;
                    bot.pathfinder.setGoal(new GoalNear(playerX, playerY, playerZ, RANGE_GOAL));
                });

            });



            bot.on('spawn', () => resolve(bot));

            bot.on('error', (err) => reject(err));


            setTimeout(() => reject(Error('Took Too Long To Spawn')), 5000);
        }, config.interval * ix);
    });
}


async function main() {



    const file = await readFile(config.file);
    const accounts = file.split(/\r?\n/).map(login => login.split(':'));
    const botProms = accounts.map(makeBot);
    const bots = (await Promise.allSettled(botProms)).map(({ value, reason }) => value || reason).filter(value => !(value instanceof Error));
    console.log(`Bot (${bots.length} / ${accounts.length}) Successfully Logged In`);



}
main();`

**Comment by u9g** [comment]
what is the format of the proxy file?

**Comment by TopZoozle** [comment]
> what is the format of the proxy file?

Host:Port

**Comment by u9g** [comment]
so the idea is that you go down the line of proxy file and login file and for i in lines of the login file use login line i and proxy line i for each login?

**Comment by u9g** [comment]
should close, come to the discord if you want help
